### DISCUSSION TITLE
Automatic updating holiday ICS files

### DISCUSSION THREAD

**Discussion by tswistak** [FossifyOrg/General-Discussion#423] [Ideas]
In SMT, holiday calendars were crowdsourced by people, what I remember was leading to people complaining that there are no current holidays for their countries, and they had to regularly provide new ICS files. I even created 2 years ago a feature request to complete remove ICS files (https://github.com/SimpleMobileTools/Simple-Calendar/issues/1533) and base on autogenerated holidays, but now I have a bit simpler idea for that, that may be sufficient.

I think, maybe it would be a good idea to generate these ICS files for all supported countries every year? As I found out, Thunderbird currently does something like this. They have a [Python script](https://github.com/thunderbird/thunderbird-website/blob/master/build_calendar.py) using [Calendarific API](https://calendarific.com/) to do this. All their generated calendars are here: https://www.thunderbird.net/en-US/calendar/holidays/.

I was looking for some resources and found out free scripts that could generate holiday ICS for us, without using Calendarific (like Thunderbird):
- https://github.com/commenthol/date-holidays-ical - NodeJS script that does exactly this.
- https://github.com/vacanza/python-holidays - Python library with holidays, we would just need to write saving to ICS.

We could generate every year new ICS files and add them with an update to APK files. Even if those ICS files would have some wrong dates (e.g. some new holidays are missing), it will be much better to have like one missing date than all dates. Also, if we had a script for this, we could even schedule it with GitHub Actions to run once a year. 

If you'd like this approach, I can try to help with that. I could write a script to automatically populate ICS files in `src/main/assets` using date-holidays-ical. I also have some experience with GitHub Actions, so I could try to write a cron action which would run once a year (let's say 1st December) to create holidays for the next year and upload them to the repo.

Alternatively, you could download the holidays from the Thunderbird site I've linked earlier. As I found correctly, they are on [CC BY-SA 3.0 license](https://www.mozilla.org/en-US/foundation/licensing/website-content/), so there wouldn't be problems with that, we just need to attribute authors.

**Comment by naveensingh** [comment]
> I was looking for some resources and found out free scripts that could generate holiday ICS for us, without using Calendarific

@tswistak will that be as up-to-date as [Calendarific API](https://calendarific.com/)?

**Reply by tswistak** [reply]
Hard to tell for 100%, but date-holidays-ical bases on https://github.com/commenthol/date-holidays which is regularly updated.

**Reply by tswistak** [reply]
@naveensingh Is approach with using date-holidays OK for you? If so, I can start work on that.

**Reply by naveensingh** [reply]
@tswistak I'm running a little busy lately and I haven't investigated these options yet. It looks like Calendarific might have more up-to-date information but it'll probably be more work to integrate.

Given the current manual approach, I think anything automated would be better so please go ahead and work on it.

**Comment by maverick74** [comment]
> @naveensingh Is approach with using date-holidays OK for you? If so, I can start work on that.

Any news on this?

**Reply by naveensingh** [reply]
@maverick74 there's already a PR: https://github.com/FossifyOrg/Calendar/pull/152

I'm going to review and merge it soon (probably later today).

**Comment by naveensingh** [comment]
Welp, the autoclose feature doesn't work for discussions.
